onSuccess fixes

lissavxo opened the pull request:
Related to #375  #374 

<!--
Depends on
---
- [ ] #
--> 

<!-- Non-technical -->
Description
---
This PR will introduce these fixes: 
- fix set success to true in paymentDialog
- fix calling onSuccess when no amount set 
- Set amount to number in onSuccess callback 



Test plan
---
Open the project, run `yarn build`  and test if onSuccess function is called and if it returns the amount properly

@lissavxo

<!-- Uncomment below to add any remarks, technical or not -->
<!-- 
Remarks
---
-->

Klakurka commented:
Why is removing the big number lib necessary here? What was wrong with it?

Klakurka reviewed (changes requested):
I think we talked about this before but either I've changed my mine or didn't really think about it clearly in the first place but until we have a dedicated non-success "transaction" toast (and success chime sound), let's block the toast if the payment ID doesn't match.

lissavxo commented:
> Why is removing the big number lib necessary here? What was wrong with it?

We removed BigNumber from form the Amount param that was being sent in Onsucces function. Now we are returning a number instead. 
 
BigNumber  returns a object so we changed it to be a number instead to have only the amount sent.

we can also adjust to try to extract the number from the BigNumber object, so it matches the value we are showing in the toast.

Klakurka reviewed (changes requested):
- The 'txid' that's provided in the onSuccess function is... I'm not really sure what it is exactly. It should be the actual broadcasted txid.
- It would be nice to have the Payment ID be available too alongside txid and amount.
- Let's have the success toast only fire using the same logic as the check mark.

Klakurka commented on docs/README.md:
```diff
@@ -525,6 +525,12 @@ successText = "Thanks!"
 
 ?> This parameter is optional. Default value is empty. Possible values are any defined function.
 
+#### *callback* arguments
+
+- hash: *string* - the hash of the broadcasted transaction
+- amount: *string* - how amount money is being requested
+- paymentId: *string* - the payment unique identifier
+
```
After actually looking at the data payload coming from the server and thinking about this for a bit, I've had a change of opinion on this.

Why not just have only the tx object be available here and they can dig into it and use whatever they need from that?

This way, we wouldn't have to go back in and add/remove/update the arguments when they change, right?

Klakurka replied:
I made this for the server as we're sending a lot of useless data to the client: https://github.com/PayButton/paybutton-server/issues/811 -- maybe we can think about updating the docs to include all of the tx data once that's eventually done? I'm not really concerned about having all of this in the docs beforehand since we made decide to keep/remove other stuff in the payload that I'm not thinking about.

Klakurka replied:
An argument against my idea here of just passing `tx` for them to use is that this is really only applicable to `on-success` and `on-transaction` but not `on-open` and `on-close`.

Thoughts?

Klakurka replied:
Estevao is good with this change so let's go ahead with it in this branch and update the docs again.

Klakurka commented on react/src/components/Widget/WidgetContainer.tsx:
```diff
@@ -144,10 +144,14 @@ export const WidgetContainer: React.FunctionComponent<WidgetContainerProps> =
         const currencyTicker = getCurrencyTypeFromAddress(to);
-
         if (!hideToasts)
           // TODO: This assumes only bch
```
Is this an old comment or does it still have relevance?

lissavxo replied:
I believe this was added just for testing

Klakurka replied:
Let's remove it while we're in here.

Klakurka commented on docs/README.md:
```diff
@@ -525,6 +525,13 @@ successText = "Thanks!"
+
+- **transaction** (`object`): 
+  - **hash** (`string`): The hash of the broadcasted transaction
+  - **amount** (`string`): Amount of money being requested
```
I also wasn't expecting us to keep the tx 'details' in the docs here but I'm fine with keeping it.

Klakurka reviewed (changes requested):
I can't get "payment success" to happen.

Example payment URI: `ecash:qp2v7kemclu7mv5y3h9qprwp0mrevkqt9gprvmm7yl?amount=10&op_return_raw=045041590000000866d82352aac4c60b`

Klakurka reviewed (approved):
OK looks like the server was having issues.

We're still getting toasts appearing when the payment ID doesn't match but we can deal with this in another task.
